tests/utils/test_shared_process_pool.py

```diff
@@ -277,15 +295,20 @@ def test_terminate_running_tasks(shared_process_pool):
     pool.set_usage("test_stage_2", 0.3)
     pool.set_usage("test_stage_3", 0.5)
 
+    manager = mp.Manager()
+    queue = manager.Queue()
+
     tasks = []
 
-    task_num = 5
-    sleep_time = 100000
+    task_num = 10
+
     for _ in range(task_num):
-        tasks.append(pool.submit_task("test_stage_1", _time_consuming_task, sleep_time))
-        tasks.append(pool.submit_task("test_stage_2", _time_consuming_task, sleep_time))
-        tasks.append(pool.submit_task("test_stage_3", _time_consuming_task, sleep_time))
+        tasks.append(pool.submit_task("test_stage_1", _blocked_until_signaled_task, queue))
+        tasks.append(pool.submit_task("test_stage_2", _blocked_until_signaled_task, queue))
+        tasks.append(pool.submit_task("test_stage_3", _blocked_until_signaled_task, queue))
 
-    # The pool should be shutdown immediately after calling terminate() without waiting for the tasks to complete
     pool.terminate()
-    pool.join()
+
+    # No tasks have been completed since they have not been signaled yet
+    for task in tasks:
+        assert not task.done()
```

Comment:
After `terminate` is called, we want to still call `pool.join()`. Its ok if some tasks have finished here. We just want to assert that not ALL tasks finished.